core/src/main/scala/kafka/common/InterBrokerSendThread.scala

```diff
@@ -51,22 +53,34 @@ abstract class InterBrokerSendThread(name: String,
     awaitShutdown()
   }
 
-  override def doWork(): Unit = {
-    var now = time.milliseconds()
+  def sendRequest(request: RequestAndCompletionHandler): Unit = {
+    sendRequests(Seq(request))
+  }
 
-    generateRequests().foreach { request =>
+  def sendRequests(requests: Iterable[RequestAndCompletionHandler]): Unit = {
+    inboundQueue.addAll(requests.asJavaCollection)
```

Comment:
Just a quick note: you had `offer` here before which has different behavior. It probably won't matter though, since the queue is unbounded.

(I only noticed this because I had commented on the `offer` before your latest commit 😄 )

Reply:
Yeah, I was going to change this to use `add` anyway to make the expectation clearer.

Reply:
Ah, didn't see your earlier comment 👍